src/docfx/template/TemplateEngine.cs

```diff
@@ -191,8 +167,22 @@ public JObject PreprocessMetadata(JObject outputMetadata, Document file)
         public JObject TransformTocMetadata(object model)
             => TransformMetadata("toc.json.js", JsonUtility.ToJObject(model));
 
+        private IReadOnlyDictionary<string, TemplateSchema>
```

Comment:
```suggestion
        private static IReadOnlyDictionary<string, TemplateSchema>
```

Reply:
what's the benefit of this?

Reply:
To ensure this method is pure and does not access member variables